### ISSUE TITLE
GET arguments override URL path?

### ISSUE THREAD

**Issue by shaneiseminger** [sofadesign/limonade#23]
I wonder if I'm missing something as I can't be the first person to notice this, but I have a case where I need to use GET arguments, and they don't appear to be working. The best way to show the problem is by example:

```
http://www.example.com/login
```

. . . loads the route for /login fine. But when I use this URL:

```
http://www.example.com/login?return=/cart
```

I get a 404 for "(GET) /return".

I dug into it and I see where request_uri tries to get the intended URL from PATH_INFO first, then tries to parse it from the query string, then checks REQUEST_URI. So when there's a query string, it never gets to REQUEST_URI . 

AFAIK under Linux/Apache, the URL is always available in REQUEST_URI. So the following fix worked for me (I don't know if it would break things on other servers). I replaced lines 1046-1069 with the following:

``` PHP

    elseif(array_key_exists('REQUEST_URI', $env['SERVER']) && !empty($env['SERVER']['REQUEST_URI']))
    {
      $request_uri = preg_replace('/\?.+/' , '', rawurldecode($env['SERVER']['REQUEST_URI']));
      $request_uri = rtrim($request_uri, '/') .'/';
      $base_path = $env['SERVER']['SCRIPT_NAME'];
      if($request_uri."index.php" == $base_path) $request_uri .= "index.php";
      $uri = str_replace($base_path, '', $request_uri);
    }
    elseif (trim($query_string, '/') != '')
    {
      $uri = $query_string;
      $get = $env['GET'];
      if(count($get) > 0)
      {
        # exclude GET params
        $keys  = array_keys($get);
        $first = array_shift($keys);
        if(strpos($query_string, $first) === 0) $uri = $first;
      }
    }
```

Basically, I just switched the elseif blocks around and changed the way that $request_uri was being parsed.
